src/ext-test/java/org/opentripplanner/ext/realtimeresolver/RealtimeResolverTest.java

```diff
@@ -0,0 +1,481 @@
+import org.opentripplanner.model.transfer.TransferService;
+import org.opentripplanner.routing.algorithm.raptoradapter.transit.TransitLayer;
+import org.opentripplanner.routing.services.TransitAlertService;
+import org.opentripplanner.routing.stoptimes.ArrivalDeparture;
+import org.opentripplanner.transit.model.basic.Notice;
+import org.opentripplanner.transit.model.basic.TransitMode;
+import org.opentripplanner.transit.model.framework.AbstractTransitEntity;
+import org.opentripplanner.transit.model.framework.FeedScopedId;
+import org.opentripplanner.transit.model.network.GroupOfRoutes;
+import org.opentripplanner.transit.model.network.Route;
+import org.opentripplanner.transit.model.network.TripPattern;
+import org.opentripplanner.transit.model.organization.Agency;
+import org.opentripplanner.transit.model.organization.Operator;
+import org.opentripplanner.transit.model.site.AreaStop;
+import org.opentripplanner.transit.model.site.MultiModalStation;
+import org.opentripplanner.transit.model.site.RegularStop;
+import org.opentripplanner.transit.model.site.Station;
+import org.opentripplanner.transit.model.site.StopLocation;
+import org.opentripplanner.transit.model.site.StopLocationsGroup;
+import org.opentripplanner.transit.model.timetable.Trip;
+import org.opentripplanner.transit.model.timetable.TripIdAndServiceDate;
+import org.opentripplanner.transit.model.timetable.TripOnServiceDate;
+import org.opentripplanner.transit.service.TransitService;
+import org.opentripplanner.updater.GraphUpdaterStatus;
+
+class RealtimeResolverTest {
+
+  private static MockLeg MOCK_LEG = new MockLeg();
+
+  private static class MockLeg implements Leg {
+
+    @Override
+    public boolean isTransitLeg() {
+      return true;
+    }
+
+    @Override
+    public boolean hasSameMode(Leg other) {
+      return false;
+    }
+
+    @Override
+    public ZonedDateTime getStartTime() {
+      return ZonedDateTime.now();
+    }
+
+    @Override
+    public ZonedDateTime getEndTime() {
+      return ZonedDateTime.now();
+    }
+
+    @Override
+    public double getDistanceMeters() {
+      throw new RuntimeException("not implemented");
+    }
+
+    @Override
+    public Place getFrom() {
+      throw new RuntimeException("not implemented");
+    }
+
+    @Override
+    public Place getTo() {
+      throw new RuntimeException("not implemented");
+    }
+
+    @Override
+    public LineString getLegGeometry() {
+      throw new RuntimeException("not implemented");
+    }
+
+    @Override
+    public int getGeneralizedCost() {
+      throw new RuntimeException("not implemented");
+    }
+
+    @Override
+    public LegReference getLegReference() {
+      return new MockLegReference();
+    }
+  }
+
+  private static class MockLegReference implements LegReference {
+
+    @Override
+    public Leg getLeg(TransitService transitService) {
+      return MOCK_LEG;
+    }
+  }
+
+  private static class MockTransitService implements TransitService {
```

Comment:
No need to mock this. You can just instantiate a default service with `new DefaultTransitService(new TransitModel())`

Reply:
Done